static/js/modals.js

```diff
@@ -64,19 +95,20 @@
       if (typeof open === "undefined") {
         open = modal.style.display === "none";
       }
-
+      
       if (open) {
         currentDialog = modal;
         modal.style.display = "flex";
-        focusFirstDescendant(modal);
         focusAfterClose = sourceEl;
         document.addEventListener("focus", trapFocus, true);
+        updateHash(triggeringHash);
       } else {
         modal.style.display = "none";
         if (focusAfterClose && focusAfterClose.focus) {
           focusAfterClose.focus();
         }
         document.removeEventListener("focus", trapFocus, true);
+        updateHash(triggeringHash);
         currentDialog = null;
```

Comment:
When I close the modal it doesn't remove the hash.

If you look at e.g. https://ubuntu.com/aws, when you close the modal it'll remove it, because it's kind of like state management, it indicates whether the modal is opened or not.

Reply:
@carkod updated to remove hash when the modal is closed